src/components/TransformWorkflow/FlowEditor/Components/OperationPanel/Forms/CastColumnOpForm.tsx

```diff
@@ -197,7 +204,16 @@ const CastColumnOp = ({
     return index == -1 ? 0 : index;
   };
 
+  /**
+    So operation nodes can be dummy (not yet saved to db) or real (saved in db).
+    Both have nodeId, but the dummy nodes have isDummy= true field. 
+    So we dont call any api in that case.
+   **/
   useEffect(() => {
+    if (node?.type === SRC_MODEL_NODE) {
+      sourceModelNodeRef.current = node; //saving table's id
+    }
+    if (node?.data.isDummy) return;
```

Comment:
this works for cast operation but not others. Also this doesn't handle the case when the node is in the middle of the chain

Try this for a table -> create a drop operation and save -> add function -> select the filter operation , you will the see columns in drop down to filter on -> click on the dummy node of filter -> check the columns in filter operations's dropdown again.